js/ui/windowManager.js

```diff
@@ -142,28 +142,143 @@ var DisplayChangeDialog = new Lang.Class({
+    },
+
+    createActorPrivate: function(actor) {
+        let effect = new EOSShellWobbly({ bridge: this });
+        actor.add_effect_with_name('endless-animation-wobbly', effect);
+        return effect;
+    }
+});
+
+// GSettingsShellWobblyEffect
+//
+// A subclass of ControllableShellWobblyEffect that gets
+// its configuration from GSettings as opposed to an external
+// caller.
+const GSettingsShellWobblyEffect = new Lang.Class({
+    Name: 'GSettingsShellWobblyEffect',
+    Extends: ControllableShellWobblyEffect,
+
+    vfunc_get_name: function() {
+        return 'gsettings-wobbly';
+    },
+
+    _init: function(params) {
+        this.parent(params);
+
+         let binder = ((key, prop) => {
+             global.settings.bind(key, this, prop, Gio.SettingsBindFlags.GET);
+         });
+
+         // Bind GSettings to effect properties
+         binder('wobbly-spring-k', 'spring-k');
+         binder('wobbly-spring-friction', 'friction');
+         binder('wobbly-slowdown-factor', 'slowdown-factor');
+         binder('wobbly-object-movement-range', 'object-movement-range');
+    }
+});
+
+// EOSShellWobbly
+//
+// Private data that exists for the attached ControllableShellWobblyEffect
+// on an AnimationsDbus.ServerSurfaceBridge (eg, a ClutterActor). It is
+// a subclass of EndlessShellFX.Wobbly (eg, a ClutterEffect) and binds to
+// the properties of the passed ControllableShellWobblyEffect.
 var EOSShellWobbly = new Lang.Class({
     Name: 'EOSShellWobbly',
     Extends: EndlessShellFX.Wobbly,
+    Implements: [ AnimationsDbus.ServerSurfaceAttachedEffect ],
+    Properties: {
+        'bridge': GObject.ParamSpec.object('bridge',
+                                           '',
+                                           '',
+                                           GObject.ParamFlags.READWRITE |
+                                           GObject.ParamFlags.CONSTRUCT_ONLY,
+                                           ControllableShellWobblyEffect)
+    },
 
     _init: function(params) {
         this.parent(params);
 
         let binder = ((key, prop) => {
-            global.settings.bind(key, this, prop, Gio.SettingsBindFlags.GET);
+            this.bridge.bind_property(key, this, prop, GObject.BindingFlags.DEFAULT);
         });
 
         // Bind to effect properties
-        binder('wobbly-spring-k', 'spring-k');
-        binder('wobbly-spring-friction', 'friction');
-        binder('wobbly-slowdown-factor', 'slowdown-factor');
-        binder('wobbly-object-movement-range', 'object-movement-range');
+        binder('spring-k', 'spring-k');
+        binder('friction', 'friction');
+        binder('slowdown-factor', 'slowdown-factor');
+        binder('object-movement-range', 'object-movement-range');
+    },
+
+    activate: function(event, detail) {
+        switch (event) {
+            case 'move':
+                detail.grabbed ? this._grabbedByMouse() : this._ungrabbedByMouse();
+                return true;
```

Comment:
Nitpick: I'd add a break here, and return true at the end of the function's body. So future additions to the switch don't have to return true by default all the time.

Reply:
Added